plugin/core/windows.py

```diff
@@ -254,6 +254,7 @@ def _needed_config(self, view: sublime.View) -> Optional[ClientConfig]:
         return None
 
     def start_async(self, config: ClientConfig, initiating_view: sublime.View) -> None:
+        config = ClientConfig.from_config(config, {})
```

Comment:
With this, I guess the line

```
resolved = config.resolve(variables)
```

can now become

```
config.resolve(variables)
```

where it mutates itself?

Reply:
I'm thinking that even the copied config should always stay unresolved and we would only resolve (and by that I mean only expanding the settings) at the time of sending them to the server. So plugins would always receive "unresolved" values (as is the case for the `_plugin.on_workspace_configuration` call already, for example.

The part of `resolve` that resolves settings for the transport would be a separate function or something.